rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildConfigurationEndpointImpl.java

```diff
@@ -178,8 +180,12 @@ public Build triggerRevision(int id, int rev, BuildParameters buildParams) {
 
     @Override
     public BuildConfiguration restoreRevision(int id, int rev) {
-        // TODO: implement
-        throw new UnsupportedOperationException("Not implemented yet");
+         Optional<BuildConfiguration> buildConfiguration = buildConfigurationProvider.restoreRevision(id, rev);
+         if (buildConfiguration.isPresent()) {
+             return buildConfiguration.get();
+         } else {
+             throw new NotFoundException("BuildConfigurationAudited with [id=" + id + ", rev=" + rev + "] does not exists");
```

Comment:
You could also use Optional orElseThrow instead of if-else.

Reply:
OK, changed.